### PR TITLE
[Release] Milestone M155

### PR DESCRIPTION
**Pull request by derrickstolee**
* 1386    GVFS Health Feature

## Mac

* 1176	Platform native notifications PR_2
* 1193	Mac ProjFS: Only hydrate/expand when deleting due to rename
* 1202	Platform native notifications PR_3
* 1215	Remove Placeholder after calling PreDelete
* 1250	Mac: Switch on the UTF8 Tests
* 1257	Mac: Bash Functional Test Hang
* 1285	Installer reliability
* 1318	Mac build scripts: consume git installer pkg directly
* 1319	Don't change version number of dev builds
* 1322	Script to generate GVFS.Installer.Mac nuspec
* 1340	PrjFSKextLogDaemon: Don't log every time a dropped kext message is detected

## Cross-Platform

* 1267	Add logging to clone/mount paths
* 1287	use repository relative paths in DiffTreeResult.TargetPath
* 1301	simplify Git functional tests of case-sensitive file paths
* 1320	Making max pipe length platform-dependant
* 1375	Remove all TODO(POSIX) and TODO(Mac) comments

## Git

* 1283	Incremental commit graph
* 1300	Update Git to include status deserialize fix
* 1342	Update Git to include tracing updates
* 1382	Update Git to include more tracing updates

## Upgrade 

* 1224	ProductUpgrader: platform specific upgrade directories
* 1225	Upgrader - platform specific directory creation
* 1228	Upgrader: copy entire application directory
* 1230	Upgrader: add command install action
* 1234	Upgrader: implement cross platform functionality
* 1268	NuGetUpgrader: fix issue with upgrader using interop services
* 1270	VFSForGit Installer non-relocatable
* 1284	Enable mac upgrade MVP
* 1297	Upgrader: NuGet Upgrader should use GitHub endpoint for notifications
* 1312	Use the new signing certificate
* 1381	Upgrade: Fix test flakiness around upgrade reminders

## FastFetch

* 1291	use correct checkout thread maximum in FastFetch

## Polish

* 1241	Update heartbeats for folder placeholders and file hydration
* 1257	Mac: Bash Functional Test Hang
* 1263	Log Folder Placeholders Removed
* 1281	PrefetchStep: don't send post-fetch request if no new packs
* 1282	Remove requirement to run --no-renames
* 1317	Setup.iss: Update GitHub URL
* 1328	Move Enlistment Directory For Functional tests to ~/GVFS.FT
* 1380	GitProcess: Catch InvalidOperationException when setting priority class

## Backports from M153

* 1259	Update Git to v2.22.0
* 1276	POSIX: Switch to Process.Start for launching GVFS.Mount
* 1278	Fix performance regression in LibGit2RepoInvoker

### CODE REVIEW

**Comment by derrickstolee** [comment]
Large Repo Perf run succeeded with expected results. All difference was within error bounds.

File: GVFS/GVFS.Platform.Mac/MacProductUpgraderPlatformStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            if (this.FileSystem.DirectoryExists(upgradeApplicationDirectory) &&
+                !this.FileSystem.TryDeleteDirectory(upgradeApplicationDirectory, out deleteDirectoryException))
+            {
+                error = $"Failed to delete {upgradeApplicationDirectory} - {deleteDirectoryException.Message}";
+
+                this.TraceException(deleteDirectoryException, nameof(this.TryPrepareApplicationDirectory), $"Error deleting {upgradeApplicationDirectory}.");
+                return false;
+            }
+
+            this.FileSystem.CreateDirectory(upgradeApplicationDirectory);
```
Comment (wilbaker) [inline GVFS/GVFS.Platform.Mac/MacProductUpgraderPlatformStrategy.cs]: @alameenshah, @jamill should we be calling  `TryCreateDirectory` here instead (or catching exceptions that `CreateDirectory` might throw)?

As best I can tell the caller of `TryPrepareApplicationDirectory` does not expect it to throw.

File: GVFS/GVFS.Platform.Mac/MacProductUpgraderPlatformStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            if (this.FileSystem.DirectoryExists(directory) &&
+                !this.FileSystem.TryDeleteDirectory(directory, out deleteDirectoryException))
+            {
+                error = $"Failed to delete {directory} - {deleteDirectoryException.Message}";
+
+                this.TraceException(deleteDirectoryException, nameof(this.TryPrepareDownloadDirectory), $"Error deleting {directory}.");
+                return false;
+            }
+
+            this.FileSystem.CreateDirectory(directory);
```
Comment (wilbaker) [inline GVFS/GVFS.Platform.Mac/MacProductUpgraderPlatformStrategy.cs]: @alameenshah @jamill same comment here.

I think we don't have to hold up the release for this, but if you agree can you file a work item or create a PR into master to update this code?

**Pull request merged**